### PR TITLE
Fix issue where setLogger did not adhere to the LoggerAwareInterface and change minimum PHP requirement to PHP 8

### LINKED ISSUE #273: Upgrade from 3.0.0 to 3.1.0 issue with psr/log requirement
Upgrading from 3.0.0 to 3.1.0 has for consequence to upgrade psr/log from 1..4 to 3.0.0 which throw an error as the setLogger method is not compatible with the upgrade version of the LoggerAwareInterface:

Declaration of Mautic\Api\Api::setLogger(Psr\Log\LoggerInterface $logger) must be compatible with Psr\Log\LoggerAwareInterface::setLogger(Psr\Log\LoggerInterface $logger): void

### PR DESCRIPTION
**Pull request by Rocksheep**
The setLogger method on the Api class never adhered to the setLogger method of LoggerAwareInterface. This is causing issues when using a newer version of the psr/log libraries which uses return types.

While testing I noticed that there was a variable being used that was not declared, so I swapped it out with false. Let me know if this is alright with you guys.

This PR fixes #273

### CODE REVIEW

**Comment by Rocksheep** [comment]
The CLA is signed!

**Comment by RCheesley** [comment]
@cla-bot check

Thanks for the PR @Rocksheep and welcome to the community!

File: lib/Api/Api.php [hunk 1]
```diff
@@ -121,14 +121,10 @@ public function getLogger()
 
     /**
      * Sets a logger.
-     *
-     * @return $this
      */
-    public function setLogger(LoggerInterface $logger)
+    public function setLogger(LoggerInterface $logger): void
```
Comment (escopecz) [inline lib/Api/Api.php:125]: This is a breaking change and will require a major release of this library.

File: tests/Api/MauticApiTestCase.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ protected function getAuth()
         $auth    = $apiAuth->newAuth($this->config, $authMethod);
         if ('BasicAuth' != $authMethod) {
             if (empty($this->config['refreshToken']) && !$auth->isAuthorized()) {
-                $this->assertTrue($authorized, 'Authorization failed. Check credentials in local.config.php.');
+                $this->assertTrue(false, 'Authorization failed. Check credentials in local.config.php.');
```
[the remark this answers is not in the document]
Comment (Rocksheep) [reply tests/Api/MauticApiTestCase.php:46]: That is a lot better actually. I did not know that a `fail` method was available

File: lib/Api/Api.php (hunk 1 above)
Comment (Rocksheep) [reply lib/Api/Api.php:125]: What needs to be done to turn this into an official major release?

Comment (escopecz) [reply lib/Api/Api.php:125]: We should review and test all the open PRs in this repo whether they should be released too once we'll make a major release so we don't have to do it soon again. We should update all the dependencies while doing so. Would you have some spare time to help with that?

Comment (Rocksheep) [reply lib/Api/Api.php:125]: I can definitely help with updating the dependencies of the project. Are there any specific versions of PHP you guys have in mind?

Comment (Rocksheep) [reply lib/Api/Api.php:125]: And just to be sure if this really is a breaking change or a bug, the interface that this class is implementing always had [void](https://github.com/php-fig/log/commit/f960c17b71ec7067eaa372f4712219a3c20e5926) (or [null](https://github.com/php-fig/log/commit/af29a61d7191c313b0db21f204a06446e62bf863) if you count the earliest version of the interface) as a return type.

**Comment by Rocksheep** [comment]
I have updated the minimum PHP version to 8.0 for now and PHP CS fixer has been updated as well. It was the only dependency I found that really needed an update.

**Comment by hugronaphor** [comment]
So, since php 7.4 is end of life. Should a new release be made to include this change?

**Comment by escopecz** [comment]
Yes, since we plan to release a new major version, we can ditch PHP 7.4 support.

**Comment by Hargalaten** [comment]
Hey there, any news regarding this PR?
Will it be merged or we have to wait for the next major?
We actually need to update our prod env and we're stuck because of this issue. 

Thanks

**Comment by christian-beckmann** [comment]
This shows indead a lack of development here, even if someone signs and has a proven PR, it keeps stucked. Why don't they proceed?

**Comment by christian-beckmann** [comment]
To use the fork, just add this to your composer.json

`"repositories": [
    {
        "type": "vcs",
        "url": "https://github.com/Rocksheep/api-library.git"
    }
],
"require": {
    "mautic/api-library": "dev-upgrade-to-php8",
}`

**Review by PierreAmmeloot — APPROVED** [review]
I have check on my side (PHP 8.2.0).

**Comment by escopecz** [comment]
Yes, once the conflicts are resolved

**Comment by Rocksheep** [comment]
I'll resolve the conflicts!

**Comment by RCheesley** [comment]
@all-contributors please add @Rocksheep for code

**Comment by RCheesley** [comment]
@all-contributors please add @PierreAmmeloot for userTesting

**Review by escopecz — APPROVED** [review]
I see no issue with these changes 👍

**Comment by escopecz** [comment]
Although the CI might need more work after the CS Fixer update...

**Comment by Rocksheep** [comment]
I'll fix the CS script and the resulting issues. So give me a few minutes and then it should all work! It's mostly some untyped `@params` in the docs and public constants that don't have the public keyword.

**Review by PierreAmmeloot — APPROVED** [review]

**Comment by Rocksheep** [comment]
All the style changes have been fixed and the action should be configured correctly now.

**Pull request merged**